File: Views/src/app/_components/administrative/local/out-sourced/dto/collect-deliver-dto.ts
import { ClientDto } from "../../../client/dto/client-dto";
import { DestinyCollectDeliverDto } from "./destiny-collect-deliver-dto";
import { PartnerDto } from "./partner-dto";
import { SourceCollectDeliverDto } from "./source-collect-deliver-dto";


export class CollectDeliverDto {
  id: number;

  transporterNoregisterd: string;

  transporterId: number;
  transporter: PartnerDto;

  //SOURCE
  sourceClientId: number;
  sourceClient: ClientDto;
  sourcePartnerId: number;
  sourcePartner: PartnerDto;
  sourceNoRegisterName: string;
  sourceNoRegisterAddress: string;

  //DESTINY
  destinyClientId: number;
  destinyClient: ClientDto;
  destinyPartnerId: number;
  destinyPartner: PartnerDto;
  destinyNoRegisterName: string;
  destinyNoRegisterAddress: string;

  start: Date;
  price: number;
  items: string;
  comments: string;
}

File: Services/Dto/Outsourced/CollectDeliverDto.cs
using System;

namespace Services.Dto.CollectsDelivers
{
    public class CollectDeliverDto
    {
        public int Id { get; set; }
        public string Subject { get; set; }
        public bool Collect { get; set; }
        public bool Deliver { get; set; }
        public DateTime Start { get; set; }
        public Decimal Price { get; set; }
        public string ItemsCollected { get; set; }
        public string ItemsDelivered { get; set; }
        public string Comments { get; set; }
        public string TransporterNoregisterd { get; set; }
        
        public int? TransporterId { get; set; }
        public virtual PartnerDto Transporter { get; set; }

        public int? CustomerId { get; set; }
        public virtual CustomerDto Customer { get; set; }

        public int? PartnerId { get; set; }
        public virtual PartnerDto Partner { get; set; }

        public int? CompanyId { get; set; }
        public virtual CompanyDto Company { get; set; }
        public string NoRegisterName { get; set; }
        public string NoRegisterAddress { get; set; }
    }
}